src/annotator/guest.js

```diff
@@ -146,44 +144,39 @@ export default class Guest extends Delegator {
       }
     });
 
-    this.plugins = {};
-
     /** @type {Anchor[]} */
     this.anchors = [];
 
+    // Setup the document type-specific integration consisting of metadata extraction,
+    // anchoring module and logic to respond to activity (eg. scrolling) in the page.
+    //
+    // In future the plan is to put these behind a common interface so that the
+    // `Guest` class doesn't need different code paths depending on the document type.
+
+    // nb. The `anchoring` field defaults to HTML anchoring and in PDFs is replaced
+    // by `PDFIntegration` below.
+    this.anchoring = anchoring;
+    this.documentMeta = new DocumentMeta(this.element);
+    if (config.documentType === 'pdf') {
+      this.pdfIntegration = new PDFIntegration(this.element, this);
```

Comment:
The way that `PDFIntegration` immediately overwrites the `this.anchoring` property of the `Guest` that is passed in is roundabout, but fixing that is out of scope for this PR.